### ISSUE TITLE
Pre-tiled geotiff as data in coverage ?

### ISSUE THREAD

**Issue by Mikiya83** [EOxServer/eoxserver#426]
Hi
Is there a way to register a pretiled image (a geotiff tiled by gdal_retile for example) as coverage ?
I tried with or without index file, without success. I also tried flat or pyramidal tiling, without success.
I thought Mosaics is the right option but i can't find a way to use it for my use case.

MapServer seems to be able to load a layer with index file and a tiled image but i can't find a way in EOX.

Thanks

**Comment by constantinius** [comment]
Hi @Mikiya83 

Thanks for your interest in EOxServer!

For this to support pre-tiled data, I think there are several options. One is to register one image for each of the highest resolution tiles as coverages, then create a mosaic and add all coverages to it.

Another solution would be to create a VRT of the pyramid and register that as a single coverage. 

Unfortunately, there is no out-of-the-box way to deal with tiled data, I'm sorry to say.

Please let us know if you had success with either approach or if you need further input!

**Comment by Mikiya83** [comment]
Thanks for your reply !
I tried the second option, which seems to be better in my case.

I can now import a VRT file which is a flat mosaic of tif files, it works well.
But i need to support pyramidal tiling (because of image size), and i can't succeed for now.
i tried this : 
- gdal_retile.py to generate multiple zoom layers (in multiple directories "1", "2", ...)
- gdaltindex for each zoom levels to generate "shp" index files
- gdalbuildvrt to generate a VRT file from all the "shp" index files
- import this VRT file as coverage

I can see an image but zooming does not improve the resolution so it's blurry. I think = i"m missing a step or an option to generate a VRT file that include some "zooming" data.

**Comment by Mikiya83** [comment]
Another idea : i generated internal overviews in a GTiff with gdaladdo, so do EOx use them to render image when client zoom/unzoom ? Goal is to reduce network impact when big TIF are managed.

**Comment by constantinius** [comment]
Yes, as EOxServer is built upon GDAL and MapServer overviews should be used when appropriate. I think there is an option to tell the VRT to use the lower zoom levels as-well, but I'm not sure whether this works with pyramids of data.
